### ISSUE TITLE
Link to v3/v4 download page should retain platform

### ISSUE THREAD

**Issue by JohnVeness** [godotengine/godot-website#590]
## Steps to reproduce
1. Use a web browser on Linux
2. Visit the v4 download page (https://godotengine.org/download/linux/)
3. Click on "Looking for other platforms? See below!"
4. Click on Windows, which takes you to https://godotengine.org/download/windows/
5. Click on "For the LTS version, Download Godot 3"

## Current behavior
You end up on URL https://godotengine.org/download/3.x/linux

## Expected behavior
You end up on URL https://godotengine.org/download/3.x/windows

## Notes
This presumably also happens with any other combination of OSes, when you're trying to download a version of the engine for a different OS than you are currently using to browse. Similar also happens if you're currently on a 3.x download page for a different OS and you click on the "For the latest version, Download Godot 4" link - again, you get switched to the OS for your current web browser, not the alternative OS.

**Comment by YuriSizov** [comment]
Yes, this is by design and is explained in the PR that changed the page.

**Comment by JohnVeness** [comment]
Sorry, I've had a good look through PRs but cannot find the explanation you mention. Can you point me in the right direction?

**Comment by YuriSizov** [comment]
Sorry, I guess this was on chat instead. So basically, yes, what you propose would be the best option. But the issue is that not all platforms are present between versions (no dedicated server/headless in 4.0), and we have no good way to indicate that. So as a simple and straightforward solution that would work consistently we drop you to the "default" page for your current device, which always exists.

If there is a solution that can account for these limitations, we can implement it. But the current way was the safest choice for the time being.

**Comment by JohnVeness** [comment]
Thanks for the explanation. Yes, I can see that it is hard to code an automatic fix for this.

It's not very elegant, but maybe each file in `collections/_download/` and `collections/_download_3/` could contain a metadata entry to point to the nearest corresponding file in the other version, e.g. a field called "platform-v3" (for files in `_download`) and "platform-v4" (for files in `_download_3`). In other words, this would be manually maintained rather than try and create an autodetection mechanism.
